src/components/Button/Button.tsx

```diff
@@ -68,44 +69,45 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       rightIcon && (
         <span
           className={cn(iconClassName, styles.icon, styles.right, {
-            [styles.noText]: !text || textIsHidden
+            [styles.noText]: noText
           })}
         >
           {rightIcon}
         </span>
       )
 
+    const getAriaLabel = () => {
+      if (ariaLabelProp) return ariaLabelProp
+      else if (hasText && typeof text === 'string') return text
```

Comment:
Doesn't this exclude if `textIsHidden`? Shouldn't it just be `else if (text) return text`?

Reply:
yeah i updated it to more acurately reflect the case where the text is hidden and also text is type string